### PR TITLE
Pause menu

### LINKED ISSUE #137: Create Menu

### PR DESCRIPTION
**Pull request by Dezerack**
Closes #137

### CODE REVIEW

File: Assets/Scripts/Weapons/LookAtMouse.cs [hunk 1]
```diff
@@ -11,8 +11,11 @@ public class LookAtMouse : MonoBehaviour
...
     {
-        Vector3 mousePosition = Camera.ScreenToWorldPoint(Input.mousePosition);
-        Vector3 relativePoint = transform.position - mousePosition;
-        transform.rotation = Quaternion.Euler(0, 0, Mathf.Atan2(relativePoint.y, relativePoint.x) * Mathf.Rad2Deg + 90);
+        if(!PauseMenu.GameIsPaused)
+        {
+            Vector3 mousePosition = Camera.ScreenToWorldPoint(Input.mousePosition);
+            Vector3 relativePoint = transform.position - mousePosition;
+            transform.rotation = Quaternion.Euler(0, 0, Mathf.Atan2(relativePoint.y, relativePoint.x) * Mathf.Rad2Deg + 90);
+        }
```
Comment (BrandonPacewic) [inline Assets/Scripts/Weapons/LookAtMouse.cs:19]: @Galaxy25 Is this the best way of doing this? Essentially we are putting a `isPaused` check at the beginning of each script update function. Does Unity not have a built in way of doing this?

**Comment by Galaxy25** [comment]
@Dezerack Pause the audio as well. Also the button aren't working on my end

File: Assets/Scripts/Weapons/LookAtMouse.cs (hunk 1 above)
Comment (Galaxy25) [reply Assets/Scripts/Weapons/LookAtMouse.cs]: @Galaxy25 Not each script. Most Unity scripts are based off time. Say the enemy is firing a bullet time is what telling the enemy to shoot again. The only thing which doesn't work with time is player inputs.

**Comment by BrandonPacewic** [comment]
I'm a little confused about where I should be going to test things as well.

**Comment by Galaxy25** [comment]
@BrandonPacewic Drag and drop the pause menu prefab. It wasn't saved to a scene to avoid merge conflicts

File: Assets/Scripts/SettingsMenu.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+{
+    public AudioMixer mainMixer;
+
+    public AudioMixer bulletMixer;
+
+    public AudioMixer musicMixer;
+
+    public AudioMixer soundFXMixer;
+
+    public AudioMixer enemyMixer;
```
Comment (BrandonPacewic) [inline Assets/Scripts/SettingsMenu.cs:16]: `enemyMixer` is never used. Are we planning on using it for something?

Comment (BrandonPacewic) [reply Assets/Scripts/SettingsMenu.cs]: Still a valid question.

**Review by BrandonPacewic — APPROVED** [review]

**Pull request merged**